src/listeners/MainListener.js

```diff
@@ -107,31 +108,42 @@ module.exports = class MainListener extends EventListener {
     const sw = (...s) => s.some(st => message.content.startsWith(st))
     const usedPrefix = sw(botMention, `<@!${this.user.id}>`) ? `${botMention} ` : sw(prefix) ? prefix : null
 
-    if (usedPrefix) {
-      const fullCmd = message.content.substring(usedPrefix.length).split(/[ \t]+/).filter(a => !spacePrefix || a)
-      const args = fullCmd.slice(1)
-      if (!fullCmd.length) return
-
-      const cmd = fullCmd[0].toLowerCase().trim()
-      const command = this.commands.find(c => c.name.toLowerCase() === cmd || (c.aliases && c.aliases.includes(cmd)))
-      if (command) {
-        const userDocument = this.database && await this.database.users.findOne(message.author.id, 'blacklisted')
-        if (userDocument && userDocument.blacklisted) return
-
-        const context = new CommandContext({
-          defaultPrefix: usedPrefix,
-          aliase: cmd,
-          client: this,
-          prefix,
-          message,
-          command,
-          language
-        })
+    if (!usedPrefix) return
 
-        this.log(`[35m"${message.content}" (${command.constructor.name}) ran by "${message.author.tag}" (${message.author.id}) on guild "${message.guild.name}" (${message.guild.id}) channel "#${message.channel.name}" (${message.channel.id})`, 'Commands')
-        this.runCommand(command, context, args, language)
-      }
+    const fullCmd = message.content.substring(usedPrefix.length).split(/[ \t]+/).filter(a => !spacePrefix || a)
+    const args = fullCmd.slice(1)
+    if (!fullCmd.length) return
+
+    const cmd = fullCmd[0].toLowerCase().trim()
+    const command = this.commands.find(c => c.name.toLowerCase() === cmd || (c.aliases && c.aliases.includes(cmd)))
+
+    const userDocument = this.database && await this.database.users.findOne(message.author.id, 'blacklisted')
+    if (userDocument && userDocument.blacklisted) return
+
+    if (!command) {
+      const didYouMean = await this.modules.didYouMean.isActive(guildId)
+
+      if (!didYouMean) return
+
+      const t = this.i18next.getFixedT(language)
+
+      const result = fuse(this, cmd)
+      if (!result) return
+      return message.reply(t('misc:didYouMean', { command: result }))
```

Comment:
make it reply with the prefix used like

Command not found. Did you mean **s!pusheen**?
Command not found. Did you mean **@Switchbladebot pusheen**?

Reply:
I think it gets very confusing.

Reply:
same for me, I think w/o prefix is waaay cleaner

Reply:
Nah, I think using the prefix is cleaner

Reply:
Ok, with prefix or not?